app/services/chat.ts

```diff
@@ -0,0 +1,190 @@
+import { UserService } from 'services/user';
+import { getPlatformService } from 'services/platforms';
+import { CustomizationService, ICustomizationSettings } from 'services/customization';
+import electron from 'electron';
+import { YoutubeService } from 'services/platforms/youtube';
+import url from 'url';
+import { WindowsService } from 'services/windows';
+import { $t } from 'services/i18n';
+
+export class ChatService extends Service {
+  @Inject() userService: UserService;
+  @Inject() customizationService: CustomizationService;
+  @Inject() windowsService: WindowsService;
+
+  private chatView: Electron.BrowserView;
+
+  init() {
+    this.userService.userLogin.subscribe(() => this.initChat());
+    this.userService.userLogout.subscribe(() => this.deinitChat());
+
+    if (this.userService.isLoggedIn()) this.initChat();
+  }
+
+  refreshChat() {
+    this.navigateToChat();
+  }
+
+  mountChat(electronWindowId: number) {
+    if (!this.chatView) return;
+
+    const win = electron.remote.BrowserWindow.fromId(electronWindowId);
+
+    // This method was added in our fork
+    (win as any).addBrowserView(this.chatView);
+  }
+
+  setChatBounds(position: IVec2, size: IVec2) {
+    if (!this.chatView) return;
+
+    this.chatView.setBounds({
+      x: Math.round(position.x),
+      y: Math.round(position.y),
+      width: Math.round(size.x),
+      height: Math.round(size.y),
+    });
+  }
+
+  unmountChat(electronWindowId: number) {
+    if (!this.chatView) return;
+
+    const win = electron.remote.BrowserWindow.fromId(electronWindowId);
+
+    // This method was added in our fork
+    (win as any).removeBrowserView(this.chatView);
+  }
+
+  private initChat() {
+    if (this.chatView) return;
+
+    this.chatView = new electron.remote.BrowserView({
+      webPreferences: {
+        nodeIntegration: false,
+      },
+    });
+
+    this.navigateToChat();
+    this.bindWindowListener();
+    this.bindDomReadyListener();
+
+    this.customizationService.settingsChanged.subscribe(changed => {
+      this.handleSettingsChanged(changed);
+    });
+  }
+
+  private deinitChat() {
+    (this.chatView as any).destroy();
+    this.chatView = null;
+  }
+
+  private async navigateToChat() {
+    const service = getPlatformService(this.userService.platform.type);
+    const nightMode = this.customizationService.nightMode ? 'night' : 'day';
+
+    // Youtube requires some special redirecting
+    if (service instanceof YoutubeService) {
+      const chatUrl = await service.getChatUrl(nightMode);
+      this.chatView.webContents.loadURL('https://youtube.com/signin');
+
+      this.chatView.webContents.once('did-navigate', () => {
+        this.chatView.webContents.loadURL(chatUrl);
+      });
+    } else {
+      const chatUrl = await service.getChatUrl(nightMode);
+      this.chatView.webContents.loadURL(chatUrl);
+    }
+  }
+
+  private bindWindowListener() {
+    this.chatView.webContents.on('new-window', e => {
```

Comment:
Would prefer this to be `event` or `evt`. The single char var makes the following line harder to parse, at least for me.